questdb-partitioned/results/c6a.metal.json

```diff
@@ -1,55 +1,55 @@
 {
     "system": "QuestDB (partitioned)",
-    "date": "2023-08-07",
+    "date": "2023-11-24",
     "machine": "c6a.metal, 500gb gp2",
     "cluster_size": 1,
     "comment": "Data load for partitioned tables is multi-threaded and thus more efficient.",
     "tags": ["Java", "time-series"],
-    "load_time": 283,
-    "data_size": 126686660753,
+    "load_time": 167,
+    "data_size": 95922186385,
     "result": [
-        [0.00797009,0.0026682,0.000222614],
-        [1.1466,0.0743637,0.0350691],
-        [1.33896,0.0294108,0.0242091],
-        [2.88612,0.0116978,0.0115375],
-        [6.32544,6.21946,6.12302],
-        [10.682,9.70659,9.67993],
-        [2.07087,0.0299815,0.0251919],
-        [0.212583,0.223532,0.226999],
-        [9.60538,8.88499,8.54807],
-        [10.3833,10.6572,10.2222],
-        [11.6556,7.75986,9.77087],
-        [10.7876,10.2966,9.82955],
-        [6.22005,5.68946,5.31396],
-        [15.575,20.1012,21.951],
-        [5.70846,5.55251,5.54523],
-        [9.41636,9.11746,9.692],
-        [12.9052,12.4855,12.6248],
-        [12.0557,11.936,12.4488],
-        [20.3953,20.5126,20.2717],
-        [0.0431935,0.0264225,0.0254709],
-        [71.9626,2.33681,2.75973],
-        [2.45832,2.31771,2.58355],
-        [48.1856,1.5962,1.3926],
-        [67.7625,0.0476454,0.215789],
-        [0.00885858,0.00393648,0.00325621],
-        [2.35187,2.58716,2.59413],
-        [2.01155,2.0116,2.01174],
-        [7.89161,12.5432,11.7973],
-        [96.8203,70.0744,70.5522],
-        [0.0433241,0.0477017,0.0384908],
-        [14.4465,14.1308,14.7048],
-        [25.9898,26.6603,25.2982],
-        [25.3329,25.5566,25.6693],
-        [24.2448,24.657,25.0246],
-        [24.9036,25.6605,25.2338],
-        [13.5415,13.897,13.403],
-        [0.566165,0.446011,0.473455],
-        [0.376493,0.436516,0.444282],
-        [0.019405,0.00734841,0.00715677],
-        [0.0346318,0.00728625,0.00714942],
-        [0.0225874,0.00709796,0.0185377],
-        [0.0197833,0.00725278,0.0184759],
-        [0.0191112,0.00719778,0.00711133]
+        [0.0202648,0.00036015,0.000996856],
+        [0.63999,0.0552794,0.0559657],
+        [0.185401,0.0221838,0.022647],
+        [2.76489,0.0116915,0.0100894],
+        [6.33487,5.98043,6.3105],
+        [10.9132,7.62953,7.90024],
+        [2.06128,0.0257175,0.0267857],
+        [0.187302,0.18176,0.178999],
+        [8.90364,9.11708,8.68719],
+        [10.6298,10.2335,9.64392],
+        [1.40442,1.08474,0.864241],
+        [0.864467,0.818844,0.941267],
+        [3.01232,2.76755,2.7604],
+        [6.84768,6.4617,6.18513],
+        [3.00846,2.92276,2.92005],
+        [9.27325,9.24496,9.3065],
+        [12.9493,12.6288,12.6471],
+        [12.4508,12.4553,12.2962],
+        [20.2326,19.9919,20.2253],
+        [0.0580621,0.0343657,0.0371066],
+        [71.8848,0.327761,0.276862],
+        [0.291733,0.278372,0.296456],
+        [47.8758,0.251057,0.267674],
+        [13.9825,0.020364,0.104363],
```

Comment:
I'm a little bit sceptical that Q23 could be executed in 20ms without query result caching or smth similar. is everything ok with this measurement?

Reply:
We don't cache anything except for the query execution plan, so, considering server restarts between query executions, the only cache used in hot query executions is the OS page cache. Let me explain why it takes 20ms to execute this query in QuestDB.

Q23 is this query:
```sql
SELECT * FROM hits WHERE URL LIKE '%google%' ORDER BY EventTime LIMIT 10;
```

QuestDB is a time-series database, so table data is sorted by a [timestamp column](https://questdb.io/docs/concept/designated-timestamp/) (`EventTime` in this case). Thus, the `ORDER BY EventTime` clause is basically a no-op. To execute this query we simply filter the table on [multiple threads](https://questdb.io/blog/2022/05/26/query-benchmark-questdb-versus-clickhouse-timescale/) and finish the query as soon as we find 10 rows.

Reply:
> QuestDB is a time-series database, so table data is sorted by a [timestamp column](https://questdb.io/docs/concept/designated-timestamp/)

ok, wasn't expecting that

Reply:
then I'm wondering if everything is ok with 4xlarge measurement for Q23

> [151.752,153.277,153.319]

Reply:
> then I'm wondering if everything is ok with 4xlarge measurement for Q23

Yes, that result is expected to be slow since the table there is non-partitioned (it doesn't have designated timestamp), i.e. there is no order in the on-disk data. Hence, the query engine has to do a full filter scan to process this query there.